File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-4.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    int res=0;
    public int goodNodes(TreeNode root) {
        if(root==null) return 0;
        int curr=root.val;
        dfs(root,curr);
        return res;
    }
    public void dfs(TreeNode root,int curr){
        if(root == null) return;
        if(root.val>=curr){
            res++;
            curr=root.val;
        }
        dfs(root.left,curr);
        dfs(root.right,curr);
    }
}

File: Data Structures & Algorithms/valid-binary-search-tree/submission-4.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        mi=float('-inf')
        ma=float('inf')
        def dfs(root,mi,ma):
            if not root:
                return True
            if root.val>mi and root.val<ma:
                return dfs(root.left,mi,root.val) and dfs(root.right,root.val,ma)
            else:
                return False
        return dfs(root,mi,ma)